src/main/java/com/google/sps/data/EventStorage.java

```diff
@@ -80,6 +85,12 @@ private static boolean isTextMatch(String search, String text) {
     return text.toLowerCase().contains(search.toLowerCase());
   }
 
+  private static boolean eventInRange(String start, String end, DateTimeRange range) {
+    return range == null || ((start == null || start.isEmpty() || range.getStartDate() == null 
+    || start.compareTo(range.getStartDate()) <= 0) && (end == null || end.isEmpty() 
+    || range.getEndDate() == null || end.compareTo(range.getEndDate()) >= 0));
+  }
+
```

Comment:
Optional: Looks a bit strange that this is a static method here. Maybe add this as method for the DateTimeRange class, e.g. `range.isInRange(start, end)`.